src/solvers/petsc_dm_wrapper.C

```diff
@@ -19,4 +19,68 @@
 
 #ifdef LIBMESH_HAVE_PETSC
 
+#include "libmesh/petsc_dm_wrapper.h"
+
+#include "libmesh/system.h"
+#include "libmesh/mesh_base.h"
+#include "libmesh/elem.h"
+
+namespace libMesh
+{
+
+void PetscDMWrapper::set_point_range_in_section( const System & system, PetscSection & section)
+{
+  const MeshBase & mesh = system.get_mesh();
+
+  unsigned int pstart = 2^30;
+  unsigned int pend = 0;
+
+  // Find minimum and maximum (inclusive) global id numbers on the current processor
+  // to build PetscSection. Currently, we're using the global node number as the point index.
+  // TODO: This is currently restricted to nodal dofs only!
+  //       Need to generalize to the cases where there's dofs at edges, faces, and interiors
+  //       as well as vertices. When we generalize, we must guarantee that each id() is unique!
+  //       We'll then add each edge, face, and/or interior as a new "point" in the PetscSection.
+  if (mesh.n_active_local_elem() > 0)
+    {
+      for (MeshBase::const_element_iterator el = mesh.active_local_elements_begin();
+           el != mesh.active_local_elements_end(); el++)
+        {
+          const Elem * elem = *el;
+
+          if (mesh.query_elem_ptr(elem->id()))
+            {
+              for (unsigned int n = 0; n < elem->n_nodes(); n++)
+                {
+                  // get the global id number of local node n
+                  dof_id_type node = elem->node(n);
+
+                  if (node < pstart)
+                    pstart = node;
+
+                  if (node > pend)
+                    pend = node;
+                }
+            }
+        }
+
+      // PetscSectionSetChart is expecting [pstart,pend), so pad pend by 1.
+      pend +=1;
+    }
+
+  // If we're on a processor who coarsened the mesh to have no local elements,
+  // we should make an empty PetscSection. An empty PetscSection is specified
+  // by passing [0,0) to the PetscSectionSetChart call.
+  else
+    {
+      pstart = 0;
+      pend = 0;
+    }
+
+  PetscErrorCode ierr = PetscSectionSetChart(section, pstart, pend);
```

Comment:
I need to ask @knepley if the chart really want there to be number of points = pend-pstart since we could easily have pstart = 0, pend=1e8 and have only those 2 elements on the processor.

Reply:
We usually assume that the number is consecutively increased from pstart to pend.

Reply:
OK, just finished talking to @knepley. He said it does need to be contiguous, but it doesn't need to be global. In fact, he said because the PetscSF stashes the tuple of local id, local proc id, they don't care really about the point number for solvers (which is all we're using this for). So we can manually construct our own local point numbering and it's fine as long as we're consistent when we set the dof ids for the point. I'll take a shot at working that in here before we merge.

Reply:
Oh, seriously?  That sounds just fine, then.  Thank him for me.

We're probably going to be stuck creating a lookup table, aren't we?  libMesh ids are all global, but (except for dof numbers) aren't necessarily contiguous.